tests/unit/type/list.tcl

```diff
@@ -15,16 +15,171 @@ proc write_big_bulk {size} {
+        r lpush list5 [string repeat a 500]
+        r linsert list5 after  [string repeat a 500] [string repeat b 500]
+        r rpush list5 [string repeat c 500]
+        assert_equal [string repeat b 500] [r lindex list5 1]
+        r LTRIM list5 1 -1
+        r llen list5
+    } {2}
+
+
+    test {plain node check compression3} {
+        r lpush list6 [string repeat a 500]
+        r LSET list6 0 [string repeat b 500]
+        assert_equal [string repeat b 500] [r lindex list6 0]
+        r lpush list6 [string repeat c 500]
+        r LSET list6 0 [string repeat d 500]
+        assert_equal [string repeat d 500] [r lindex list6 0]
+    }
+
+    r config set list-compress-depth 0
+
+    test {plain node check push and pop} {
+        r lpush lst 9
+        r lpush lst xxxxxxxxxx
+        r lpush lst xxxxxxxxxx
+        set s0 [s used_memory]
+        assert {$s0 > 10}
+        assert {[r llen lst] == 3}
+        set s0 [r rpop lst]
+        set s1 [r rpop lst]
+        assert {$s0 eq "9"}
+        assert {[r llen lst] == 1}
+        r lpop lst
+        assert {[string length $s1] == 10}
+    }
+
+    test {plain node check lindex and linsert} {
+        r lpush lst xxxxxxxxxxx
+        r lpush lst 9
+        r lpush lst xxxxxxxxxxx
+        r linsert lst before "9" "8"
+        assert {[r lindex lst 1] eq "8"}
+        r linsert lst BEFORE "9" "7"
+        r linsert lst BEFORE "9" "xxxxxxxxxxx"
+        assert {[r lindex lst 3] eq "xxxxxxxxxxx"}
+    }
+
+    test {plain node check LTRIM} {
+        r lpush lst1 9
+        r lpush lst1 xxxxxxxxxxx
+        r lpush lst1 9
+        r LTRIM lst1 1 -1
+        r llen lst1
+    } {2}
+
+    test {plain node check LREM} {
+        r lpush ls3 one
+        r lpush ls3 xxxxxxxxxxx
+        set s0 [s used_memory]
+        assert {$s0 > 10}
+        r lpush ls3 9
+        r LREM ls3 -2 "one"
+        r llen ls3
+    } {2}
+
+    test {plain node check LSET} {
+        r RPUSH ls4 "aa"
+        r RPUSH ls4 "bb"
+        r RPUSH ls4 "cc"
+        r write "*4\r\n\$4\r\nLSET\r\n\$3\r\nls4\r\n\$1\r\n0\r\n"
+        write_big_bulk 10;
+        set s0 [s used_memory]
+        assert {$s0 > 10}
+    }
+
+    test {plain node check LPOS} {
+        r RPUSH ls5 "aa"
+        r RPUSH ls5 "bb"
+        r RPUSH ls5 "cc"
+        r LSET ls5 0 xxxxxxxxxxx
+        r write "*4\r\n\$4\r\nLSET\r\n\$3\r\nls5\r\n\$1\r\n0\r\n"
+        write_big_bulk 10;
+        r LPOS ls5 cc
+    } {2}
+
+    test {plain node check lmove} {
+        r RPUSH {mylist} "aa"
+        r RPUSH {mylist} "bb"
+        r LSET {mylist} 0 xxxxxxxxxxx
+        r LMOVE {mylist} mylist RIGHT LEFT
+        r LMOVE {mylist} mylist LEFT RIGHT
+        r lpop mylist
+        r lpop mylist
+        r lpop {mylist}
+        r lpop {mylist}
+    } {}
+}
+
+start_server [list overrides [list save ""] ] {
     r config set proto-max-bulk-len 10000000000 ;#10gb
     r config set client-query-buffer-limit 10000000000 ;#10gb
+    set str_length 1000
```

Comment:
this should be some 5gb, right?
see more details in https://github.com/redis/redis/pull/9357#discussion_r689123060

Reply:
yes... its temporary so I can run the test on a regular machine and not an extra big one.

Reply:
ok, just remember that doing so doesn't really test what you intend to test.
i.e. it does test that the test code is correct, but it doesn't prove that there aren't any bugs in the C code (like using `int` instead of `long` or `size_t`).

Reply:
@oranagra I've added a temporary comment and will be changed when ready to submit.